### DISCUSSION TITLE
Transaction fee per byte is very high in Dogecoin, Can we change fee ?

### DISCUSSION THREAD

**Discussion by akhilccrb** [trustwallet/wallet-core#2222] [Q&A]
Here is the code used , setting byte fee to 1. But sometimes fees can't control.                   

    let input_5 = BitcoinSigningInput.with {
                          $0.hashType = BitcoinScript.hashTypeForCoin(coinType: .dogecoin)
                          $0.amount = amount_5
                          $0.byteFee = 1
                          $0.useMaxAmount = false
                          $0.toAddress = "DNGFS97bTyoxUUZKwemSiv3WhNHjbfaXTA"
                          $0.changeAddress = "DDEZXM8NVnk2fF3mGfLJKmYq4t2MteFrTA"
                          $0.utxo = utxosArrList
                          $0.coinType = CoinType.dogecoin.rawValue
                          $0.privateKey = [privateKey11.data]

                      }

               let output_5: DecredSigningOutput = AnySigner.sign(input: input_5, coin: .dogecoin)

**Comment by optout21** [comment]
1 satoshi/byte is the lowest fee possible in SigningInput.
Note that in cases when there are a high number of input UTXOs, the transaction takes many bytes, which results in higher fee.

The workaround way to set even lower fee:
- Prepare signing input with 1 sat/byte fee rate
- Invoke plan
- In the resulting plan increase the amount of change, thus reducing fee (fee is availableAmount - outputAmount - changeAmount)
- Prepare signing input with the above plan
- Invoke sign

**Reply by akhilccrb** [reply]
let input_5 = BitcoinSigningInput.with {
                          $0.hashType = BitcoinScript.hashTypeForCoin(coinType: .dogecoin)
                          $0.amount = amount_5
                          $0.byteFee = 1
                          $0.useMaxAmount = false
                          $0.toAddress = "DNGFS97bTyoxUUZKwemSiv3WhNHjbfaXTA"
                          $0.changeAddress = "DDEZXM8NVnk2fF3mGfLJKmYq4t2MteFrTA"
                          $0.utxo = utxosArrList
                          $0.coinType = CoinType.dogecoin.rawValue
                          $0.privateKey = [privateKey11.data]
                          $0.plan = BitcoinTransactionPlan.with {
                              $0.amount = amount_5
                              $0.availableAmount = sendUTXO.amount
                              $0.fee = 1
                              $0.change = amount_5
                              $0.utxos = [sendUTXO]
                          }
                      }


I added plan, but fee is different.

**Reply by optout21** [reply]
It doesn't work quite like that.  In the plan, the 3 fields `amount`, `change`, and `fee` have to add up exactly to `availableAmount`.  The transfer transaction will have two outputs: amount and change, and fee is implicit, it is the leftover, it is computed as `availableAmount - amount - change`.
If you want a special fee, you should:
- invoke Plan to come up with suggested plan
- change plan: increase the change by the diff you want, and reduce fee by the same diff
- invoke Sign with this plan in the signingInput.

In your example, amount and change is set to the same amount, which does not seem right.

**Reply by akhilccrb** [reply]
Can please specify ?
